File: clients/javascript/src/apis/cacheNamespace.ts
// this file is @generated

import {
    type CacheCreateNamespaceIn,
    CacheCreateNamespaceInSerializer,
} from '../models/cacheCreateNamespaceIn';
import {
    type CacheCreateNamespaceOut,
    CacheCreateNamespaceOutSerializer,
} from '../models/cacheCreateNamespaceOut';
import {
    type CacheGetNamespaceIn,
    CacheGetNamespaceInSerializer,
} from '../models/cacheGetNamespaceIn';
import {
    type CacheGetNamespaceOut,
    CacheGetNamespaceOutSerializer,
} from '../models/cacheGetNamespaceOut';
import { HttpMethod, CoyoteRequest, type CoyoteRequestContext } from "../request";

export class CacheNamespace {
    public constructor(private readonly requestCtx: CoyoteRequestContext) {}

    /** Create cache namespace */
        public create(
            cacheCreateNamespaceIn: CacheCreateNamespaceIn,
            ): Promise<CacheCreateNamespaceOut> {
            const request = new CoyoteRequest(HttpMethod.POST, "/api/v1/cache/namespace/create");

            request.setBody(
                    CacheCreateNamespaceInSerializer._toJsonObject(
                        cacheCreateNamespaceIn,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    CacheCreateNamespaceOutSerializer._fromJsonObject,
                );
            }

        

    /** Get cache namespace */
        public get(
            cacheGetNamespaceIn: CacheGetNamespaceIn,
            ): Promise<CacheGetNamespaceOut> {
            const request = new CoyoteRequest(HttpMethod.POST, "/api/v1/cache/namespace/get");

            request.setBody(
                    CacheGetNamespaceInSerializer._toJsonObject(
                        cacheGetNamespaceIn,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    CacheGetNamespaceOutSerializer._fromJsonObject,
                );
            }

        

    }

File: clients/java/src/main/java/com/svix/coyote/apis/CacheNamespace.java
// this file is @generated
package com.svix.coyote.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.coyote.ApiException;
import com.svix.coyote.HttpClient;
import com.svix.coyote.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.coyote.models.CacheCreateNamespaceIn;
import com.svix.coyote.models.CacheCreateNamespaceOut;
import com.svix.coyote.models.CacheGetNamespaceIn;
import com.svix.coyote.models.CacheGetNamespaceOut;

public class CacheNamespace {
    private final HttpClient client;

    public CacheNamespace(HttpClient client) {
        this.client = client;
    }

    /** Create cache namespace */
    public CacheCreateNamespaceOut create(
        final CacheCreateNamespaceIn cacheCreateNamespaceIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/cache/namespace/create");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            cacheCreateNamespaceIn,
            CacheCreateNamespaceOut.class
        );
    }

    /** Get cache namespace */
    public CacheGetNamespaceOut get(
        final CacheGetNamespaceIn cacheGetNamespaceIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/cache/namespace/get");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            cacheGetNamespaceIn,
            CacheGetNamespaceOut.class
        );
    }
}